sale_order_import/wizard/import_sale.py

```diff
@@ -0,0 +1,682 @@
+    def _default_picking_policy(self):
+        """Get picking policy"""
+        return self.env.company.picking_policy
+
+    @api.model
+    def _default_customer_invoice_journal(self):
+        """Get customer invoice journal"""
+        return self.env.company.customer_invoice_journal_id
+
+    @api.model
+    def _default_customer_payment_journal(self):
+        """Get customer payment journal"""
+        return self.env.company.customer_payment_journal_id
+
+    input_file = fields.Binary(
+        string="Sale Order File (.csv Format)",
+        required=True,
+    )
+    datas_fname = fields.Char(string="File Path")
+    picking_policy = fields.Selection(
+        [("direct", "As soon as possible"), ("one", "When all products are ready")],
+        string="Shipping Policy",
+        required=True,
+        default=_default_picking_policy,
+    )
+    customer_invoice_journal_id = fields.Many2one(
+        "account.journal",
+        string="Customer Invoice Journal",
+        required=True,
+        default=_default_customer_invoice_journal,
+    )
+    customer_payment_journal_id = fields.Many2one(
+        "account.journal",
+        string="Customer Payment Journal",
+        default=_default_customer_payment_journal,
+    )
+    asynchronous = fields.Boolean(string="Import Asynchronously", default=True)
+    process_payment = fields.Boolean("Process Payments")
+
+    @api.model
+    def _get_order_value_dict(
+        self,
+        error_vals,
+        partner_tel,
+        product_id,
+        pricelist_id,
+        warehouse_id,
+        team_id,
+        carrier_id,
+        partner_dict,
+        product_dict,
+        pricelist_dict,
+        picking_dict,
+        warehouse_dict,
+        team_dict,
+        carrier_dict,
+    ):
+        """Get order value dict"""
+        partner_value = partner_tel
+        partner_name = self._context.get("partner_name")
+        if partner_name:
+            self = self.with_context(partner_name=partner_name.strip())
+        self._get_partner_dict(partner_value, partner_dict)
+
+        product_id_value = product_id
+        self._get_product_dict(product_id_value, product_dict, error_vals)
+
+        pricelist_value = pricelist_id
+        self._get_pricelist_dict(pricelist_value, pricelist_dict, error_vals)
+
+        warehouse_value = warehouse_id
+        self._get_picking_dict(
+            warehouse_value, picking_dict, warehouse_dict, error_vals
+        )
+
+        team_value = team_id
+        self._get_team_dict(team_value, team_dict, error_vals)
+
+        carrier_value = carrier_id
+        if carrier_value:
+            self._get_carrier_dict(carrier_value, carrier_dict, error_vals)
+
+        return (
+            partner_value,
+            product_id_value,
+            pricelist_value,
+            warehouse_value,
+            team_value,
+            carrier_value,
+        )
+
+    @api.model
+    def _get_order_value(self, error_vals, taxes, taxes_id):
```

Comment:
@AungKoKoLin1997 Now, checking to make sure the numbers are positive has been removed.  Please restore this check in an efficient manner.

https://github.com/qrtl/mi7-custom/pull/234/commits/cccb13e358ffb7311619f52b48638f651a1b1918

This suggestion on base_data_import is not good as it's not generic.  We cannot assume all the float values are positive.